multy_core/src/bitcoin/bitcoin_account.cpp

```
@@ -297,5 +297,45 @@ AccountPtr make_bitcoin_account(const char* private_key)
     return AccountPtr(new BitcoinAccount(std::move(key), HDPath(), net_type));
 }
 
+BinaryDataPtr parse_bitcoin_address(const char* address,
+                                    BitcoinNetType* net_type,
+                                    enum BitcoinAddressType* address_type)
+{
+    BinaryDataPtr out_binary_data;
+    size_t binary_data_size = 0;
+    THROW_IF_WALLY_ERROR(
+                wally_base58_get_length(address, &binary_data_size),
+                "Can not get the len");
+    make_binary_data(binary_data_size, reset_sp(out_binary_data));
+    THROW_IF_WALLY_ERROR(
+                wally_base58_to_bytes(
+                    address, BASE58_FLAG_CHECKSUM,
+                    const_cast<uint8_t*>(out_binary_data->data),
+                    out_binary_data->len, &binary_data_size),
+                "Invalid address");
+    out_binary_data->len = binary_data_size;
+    if (address[0] == '1')
+    {
+        *net_type = BITCOIN_MAINNET;
+        *address_type = BITCOIN_ADDRESS_P2PKH;
+    }
+    else if (address[0] == '3')
+    {
+        *net_type = BITCOIN_MAINNET;
+        *address_type = BITCOIN_ADDRESS_P2SH;
+    }
+    else if (address[0] == 'm' || address[0] == 'n')
+    {
+        *net_type = BITCOIN_TESTNET;
+        *address_type = BITCOIN_ADDRESS_P2PKH;
+    }
+    else if (address[0] == '2')
+    {
+        *net_type = BITCOIN_TESTNET;
+        *address_type = BITCOIN_ADDRESS_P2SH;
+    }
```

At this point we don't know neither net type nor address type, it looks like we should throw an exception that we don't support this kind of addresses.